cmd/skopeo/sync.go

```diff
@@ -609,22 +674,27 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 			var destSuffix string
 			switch ref.Transport() {
 			case docker.Transport:
-				// docker -> dir or docker -> docker
-				destSuffix = ref.DockerReference().String()
+				// docker -> dir,docker,oci, trim domain
+				dockerRef := ref.DockerReference()
+				tagref, _ := reference.TagNameOnly(dockerRef).(reference.Tagged)
+				destSuffix = fmt.Sprintf("%s:%s", reference.Path(dockerRef), tagref.Tag())
+
 			case directory.Transport:
-				// dir -> docker (we don't allow `dir` -> `dir` sync operations)
+				// dir -> docker or oci -> docker(we don't allow `dir` -> `dir` sync operations)
```

Comment:
This seems incorrect — `ref` refers to the source, and `oci.Transport` is handled below.